lib/Backend/GlobOpt.cpp

```diff
@@ -13450,7 +13450,10 @@ GlobOpt::TypeSpecializeStElem(IR::Instr ** pInstr, Value *src1Val, Value **pDstV
     case ObjectType::Uint16MixedArray:
     case ObjectType::Int32MixedArray:
     Int32Array:
-        toType = TyInt32;
+        if (this->DoAggressiveIntTypeSpec() || this->DoFloatTypeSpec())
+        {
+            toType = TyInt32;
+        }
         break;
 
     case ObjectType::Uint32Array:
```

Comment:
@LouisLaf When the sym is already int/float specialized, I have not added the check to see if float/int typespec was disabled. This should be okay ?